### ISSUE TITLE
Multiline trailing comma rule flags type parameters 2

### ISSUE THREAD

**Issue by mizunashi-mana** [palantir/tslint#1905]
### Bug Report

- __TSLint version__: `4.1.1`
- __TypeScript version__: `2.1.4`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
var foo = <
  A,
  B
>(a: A, b: B) => 42;
function foo(a: [
  number,
  string
]): void {
  return 42; 
}
```

with `tslint.json` configuration:

```json
{
    "trailing-comma": [true, {"multiline": "always", "singleline": "never"}]
}
```

#### Actual behavior

```typescript
var foo = <
  A,
  B
   ~ [Missing trailing comma]
>(a: A, b: B) => 42;
function foo(a: [
  number,
  string
  ~ [Missing trailing comma]
]): void {
  return 42; 
}
```

This is invalid typescript code.  Related: #1769 

#### Expected behavior

```typescript
var foo = <
  A,
  B
>(a: A, b: B) => 42;
function foo(a: [
  number,
  string
]): void {
  return 42; 
}
```

**Comment by nchen63** [comment]
also fomr #1769, this fails:

```
function foo(a: [
  number,
  string
  ~ [Missing trailing comma] // expecting no warnings
]): void {
  return 42; 
}
```
